src/rocks_recovery_unit.cpp

```diff
@@ -373,6 +382,13 @@ namespace mongo {
         _releaseSnapshot();
     }
 
+    std::shared_ptr<RocksSnapshotManager::SnapshotHolder>  RocksRecoveryUnit::_getCommittedSnapshot() const {
+        if (!_snapshotManager->materializedCommittedSnapshot()) {
```

Comment:
Can you try moving this check into `RocksRecoveryUnit::setReadFromMajorityCommittedSnapshot()`?
I believe this won't break the code (`_getCommittedSnapshot()` is only called after `_readFromMajorityCommittedSnapshot` check right now) and we'll be able to capture majority snapshot as close as possible to the place where it's been requested.